### ISSUE TITLE
Error getting Probe form in web UI

### ISSUE THREAD

**Issue by justb4** [geopython/GeoHealthCheck#168]
This happens now and then, not always:

* add a `WMS Resource`
* in Edit mode Add a `WMS GetMap Probe`
* then  sometimes errors in UI are seen like

```
Error getting Probe form for GeoHealthCheck.plugins.probe.wms.WmsGetMapV1
Error getting Probe form for GeoHealthCheck.plugins.probe.wms.WmsGetMapV1All
```
Sometimes even the `Resource` cannot be edited. The error is not permanent. Seems to go away after some time or after restart. Suspicion is that these errors are related to ``Probe Metadata Caching`` as this only happens with `Probes` that apply MD Caching (here for WMS Caps doc) and MD is only cached for some specific time. 

Happens in `mod_wsgi` and `Docker` running envs and with `SQLIte` and `PostgreSQL` backends.

Exception in log at least two types of Exceptions:

```
[2017-10-16 14:22:53,201] ERROR in app: Exception on /probe/GeoHealthCheck.plugins.probe.wms.WmsGetMapV1/3/edit_form [GET]
Traceback (most recent call last):
  File "/venv/lib/python2.7/site-packages/flask/app.py", line 1982, in wsgi_app
    response = self.full_dispatch_request()
  File "/venv/lib/python2.7/site-packages/flask/app.py", line 1614, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/venv/lib/python2.7/site-packages/flask/app.py", line 1517, in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File "/venv/lib/python2.7/site-packages/flask/app.py", line 1612, in full_dispatch_request
    rv = self.dispatch_request()
  File "/venv/lib/python2.7/site-packages/flask/app.py", line 1598, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "/venv/lib/python2.7/site-packages/flask_login.py", line 758, in decorated_view
    return func(*args, **kwargs)
  File "/GeoHealthCheck/GeoHealthCheck/app.py", line 731, in get_probe_edit_form
    probe_obj.expand_params(resource)
  File "/GeoHealthCheck/GeoHealthCheck/plugins/probe/wms.py", line 151, in expand_params
    raise err
KeyError: 'dictionary is empty'
```

or 

```
[2017-10-16 14:29:56,958] ERROR in app: Exception on /resource/3/edit [GET]
Traceback (most recent call last):
  File "/venv/lib/python2.7/site-packages/flask/app.py", line 1982, in wsgi_app
    response = self.full_dispatch_request()
  File "/venv/lib/python2.7/site-packages/flask/app.py", line 1614, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/venv/lib/python2.7/site-packages/flask/app.py", line 1517, in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File "/venv/lib/python2.7/site-packages/flask/app.py", line 1612, in full_dispatch_request
    rv = self.dispatch_request()
  File "/venv/lib/python2.7/site-packages/flask/app.py", line 1598, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "/venv/lib/python2.7/site-packages/flask_login.py", line 758, in decorated_view
    return func(*args, **kwargs)
  File "/GeoHealthCheck/GeoHealthCheck/app.py", line 686, in edit_resource
    resource=resource, probes_avail=probes_avail)
  File "/venv/lib/python2.7/site-packages/flask/templating.py", line 134, in render_template
    context, ctx.app)
  File "/venv/lib/python2.7/site-packages/flask/templating.py", line 116, in _render
    rv = template.render(context)
  File "/venv/lib/python2.7/site-packages/jinja2/environment.py", line 1008, in render
    return self.environment.handle_exception(exc_info, True)
  File "/venv/lib/python2.7/site-packages/jinja2/environment.py", line 780, in handle_exception
    reraise(exc_type, exc_value, tb)
  File "/GeoHealthCheck/GeoHealthCheck/templates/edit_resource.html", line 1, in top-level template code
    {% extends "layout.html" %}
  File "/GeoHealthCheck/GeoHealthCheck/templates/layout.html", line 125, in top-level template code
    {% block body %}
  File "/GeoHealthCheck/GeoHealthCheck/templates/edit_resource.html", line 53, in block "body"
    {% include 'includes/probe_edit_form.html' %}
  File "/GeoHealthCheck/GeoHealthCheck/templates/includes/probe_edit_form.html", line 5, in top-level template code
    <strong>{{ probe_info.NAME }} </strong>- by  {{ probe_info.AUTHOR }}
  File "/venv/lib/python2.7/site-packages/jinja2/environment.py", line 430, in getattr
    return getattr(obj, attribute)
UndefinedError: 'dict object' has no attribute u'GeoHealthCheck.plugins.probe.wms.WmsGetMapV1'

```

**Comment by justb4** [comment]
Ok problem found: get Layers or FeatureTypes from WMS or WFS Caps doc using `OwsLib` are `OrderedDict`s. The WMS/WFS Probes used `popitem()` which is destructive. Hence OWS endpoints with just one layer very quickly gave an error! Fix upcoming.

**Comment by justb4** [comment]
Fixed and merged via #169 into master.
